py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/logging_utils.py

```diff
@@ -0,0 +1,72 @@
+import logging
+import time
+import uuid
+
+from typing import Optional, Union
+
+MdValues = Union[str, int, float]
+
+
+class ProcessLogger:
+    """
+    Class to help with logging events that happen inside of a function.
+    """
+
+    def __init__(
+        self, process_name: str, **metadata: Optional[MdValues]
+    ) -> None:
+        """
+        create a process logger with a name and optional metadata. a start time
+        and uuid will be created for timing and unique identification
+        """
+        self.process_name = process_name
+        self.uuid = uuid.uuid4()
+        self.start_time = time.time()
```

Comment:
I doubt it will really matter for our purposes, but I have read that `time.monotonic()` is specifically provided for duration measurements